src/executor/operator/physical_merge_limit.cppm

```diff
@@ -29,31 +31,32 @@ namespace infinity {
 
 export class PhysicalMergeLimit final : public PhysicalOperator {
 public:
-    explicit PhysicalMergeLimit(SharedPtr<Vector<String>> output_names,
-                                SharedPtr<Vector<SharedPtr<DataType>>> output_types,
-                                u64 id,
-                                SharedPtr<Vector<LoadMeta>> load_metas)
-        : PhysicalOperator(PhysicalOperatorType::kMergeLimit, nullptr, nullptr, id, load_metas), output_names_(Move(output_names)),
-          output_types_(Move(output_types)) {}
+    explicit PhysicalMergeLimit(u64 id,
+                                UniquePtr<PhysicalOperator> left,
+                                SharedPtr<BaseExpression> limit_expr,
+                                SharedPtr<BaseExpression> offset_expr,
+                                SharedPtr<Vector<LoadMeta>> load_metas);
 
     ~PhysicalMergeLimit() override = default;
 
     void Init() override;
 
     bool Execute(QueryContext *query_context, OperatorState *operator_state) final;
 
-    inline SharedPtr<Vector<String>> GetOutputNames() const final { return output_names_; }
+    inline SharedPtr<Vector<String>> GetOutputNames() const final { return left_->GetOutputNames(); }
 
-    inline SharedPtr<Vector<SharedPtr<DataType>>> GetOutputTypes() const final { return output_types_; }
+    inline SharedPtr<Vector<SharedPtr<DataType>>> GetOutputTypes() const final { return left_->GetOutputTypes(); }
 
     SizeT TaskletCount() override {
         Error<NotImplementException>("TaskletCount not Implement");
         return 0;
     }
 
 private:
-    SharedPtr<Vector<String>> output_names_{};
-    SharedPtr<Vector<SharedPtr<DataType>>> output_types_{};
+    SharedPtr<BaseExpression> limit_expr_{};
+    SharedPtr<BaseExpression> offset_expr_{};
+
+    SharedPtr<LimitCounter> counter_;
```

Comment:
How about UniquePtr ?

Reply:
If you use `AtomicCounter`, SharedPtr is required to ensure sharing